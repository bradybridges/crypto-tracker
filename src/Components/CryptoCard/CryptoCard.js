import React from 'react';

const CryptoCard = (props) => {
  const price = Number(props.price).toFixed(2);
  const percentChange = (Number(props.percentChange) * 100).toFixed(2);
  return (
    <section>
      <div>
        <img src={props.logo} alt='Cryptocurrency logo' />
        <h1>{props.name}</h1>
      </div>
      <p>${price}</p>
      <p>%{percentChange}</p>
      <p>Rank {props.rank}</p>
    </section>
  );
}

export default CryptoCard;