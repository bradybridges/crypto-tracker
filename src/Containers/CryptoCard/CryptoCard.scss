h4 {
  font-size: 1.5em;
  margin: 0;
  width: 25%;
}

p {
  font-size: 1.2em;
  width:25%;
}

img {
  width: 34px;
}

.crypto-card {
  color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #3e3e3ed4;
  border-radius: 4px;
  padding: 10px;
  margin-bottom: 10px;
  &:hover {
    cursor: pointer;
  }
}

.navlink {
  text-decoration: none;
}

.card-item {
  width: 30%;
}

