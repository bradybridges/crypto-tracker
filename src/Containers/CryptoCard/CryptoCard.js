import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import downArrow from '../../Images/down-arrow.png';
import upArrow from '../../Images/up-arrow.png';
import './CryptoCard.scss';

export const CryptoCard = (props) => {
  const price = Number(props.price).toFixed(2);
  const percentChange = (Number(props.percentChange) * 100).toFixed(2);
  const arrowSrc = Number(props.percentChange) < 0 ? downArrow : upArrow;
  return (
    <NavLink className="navlink" to={`/coins/${props.name}`}>
      <section className="crypto-card">
        <h4 className="card-item">{props.name}</h4>
        <p className="card-item">
          $
          {price}
        </p>
        <p className="card-item">
          %
          {percentChange}
        </p>
        <img src={arrowSrc} alt="percent change indicator" />
      </section>
    </NavLink>
  );
};
export const mapStateToProps = (state) => ({
  timeFrame: state.timeFrame,
});

export default connect(mapStateToProps)(CryptoCard);

CryptoCard.propTypes = {
  price: PropTypes.string.isRequired,
  percentChange: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};
