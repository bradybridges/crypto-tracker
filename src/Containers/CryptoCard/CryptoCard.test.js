import React from 'react';
import { shallow } from 'enzyme';
import { CryptoCard, mapStateToProps } from './CryptoCard';

describe('CryptoCard', () => {
  const wrapper = shallow(<CryptoCard price="1001.83" percentChange="-0.1345" name="Bitcoin" />);
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return timeFrame', () => {
    const mockState = {
      timeFrame: '1d',
      error: 'There was an error',
    }
    const result = mapStateToProps(mockState);
    expect(result).toEqual({ timeFrame: '1d' });
  });
});
