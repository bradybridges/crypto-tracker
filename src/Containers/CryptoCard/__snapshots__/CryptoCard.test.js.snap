// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCard should match the snapshot 1`] = `
<NavLink
  to="/coins/Bitcoin"
>
  <section
    className="crypto-card"
  >
    <h4>
      Bitcoin
    </h4>
    <p>
      $
      1001.83
    </p>
    <p>
      %
      -13.45
    </p>
    <img
      alt="percent change indicator"
      src="down-arrow.png"
    />
  </section>
</NavLink>
`;
